src/core/agent/services/identifierService.ts

```diff
@@ -140,35 +150,116 @@ class IdentifierService extends AgentService {
     };
   }
 
+  async resolvePendingIdentifier() {
+    const pendingIdentifierCreation = await this.basicStorage.findById(
+      MiscRecordId.IDENTIFIERS_PENDING_CREATION
+    );
+
+    if (
+      pendingIdentifierCreation &&
+      Array.isArray(pendingIdentifierCreation?.content?.name)
+    ) {
+      for (const name of pendingIdentifierCreation.content.name) {
+        let metadata: Omit<IdentifierMetadataRecordProps, "id" | "createdAt">;
+        const [themeString, rest] = name.split(":");
+        if (!rest) {
+          throw new Error(
+            `${IdentifierService.IDENTIFIER_METADATA_RECORD_MISSING}`
+          );
+        }
+
+        const theme = Number(themeString);
+        const groupMatch = rest.match(/^(\d)-(.+)-(.+)$/);
+        if (groupMatch) {
+          const [, initiatorFlag, groupId, displayName] = groupMatch;
+          metadata = {
+            theme,
+            displayName,
+            groupMetadata: {
+              groupId,
+              groupInitiator: initiatorFlag === "1",
+              groupCreated: false,
+            },
+          };
+        } else {
+          metadata = {
+            theme,
+            displayName: rest,
+          };
+        }
+
+        await this.createIdentifier(metadata, true);
+      }
+    }
+  }
+
   @OnlineOnly
   async createIdentifier(
-    metadata: Omit<IdentifierMetadataRecordProps, "id" | "createdAt">
-  ): Promise<CreateIdentifierResult> {
-    const startTime = Date.now();
-
+    metadata: Omit<IdentifierMetadataRecordProps, "id" | "createdAt">,
+    isResolvePendingCration?: boolean
+  ): Promise<void> {
```

Comment:
This needs to return the `Promise<CreateIdentifierResult>` I think. I think we'd need it for when we create a multi-sig